nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java

```diff
@@ -275,6 +318,12 @@ private CredentialsProvider addCredentials(final CredentialsProvider credentials
         return cp;
     }
 
+    private BasicHeader createApiKeyAuthorizationHeader(String apiKeyId, String apiKey) {
+        final String apiKeyCredentials = String.format("%s:%s", apiKeyId, apiKey);
+        final String apiKeyAuth = Base64.getEncoder().encodeToString((apiKeyCredentials).getBytes(StandardCharsets.UTF_8));
```

Comment:
Hi @nandorsoma , I ran into issues with Elastic Search API auth. Standard HTTP client works well for me, but it did not work with this client implementation. I looked into the code then found some questionable lines. What is API key id doing here? It is a required field, but should it be part of the token? And why does it encode the string once more as we already passed in base64 token string. Could you elaborate?

Reply:
The `id` and `key` would be the values returned by Elasticsearch when you [create an api key](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html#security-api-create-api-key-example) - NiFi requires both of these to construct the `Authorization` (ApiKey) header

From your description, it sounds like you are applying the `encoded` field from the Elasticsearch response (which is the Base64 encoded `id` & `key`) - NiFi doesn't currently use that in its approach - if you require that, you could raise a Jira to request it be added (submit a PR of your able) to make that an alternative to providing these existing fields

Reply:
Thanks, Chris. Feeling enlightened.